platform/commonUI/browse/src/navigation/NavigateAction.js

```diff
@@ -47,36 +44,12 @@ define(
          *          navigation has been updated
          */
         NavigateAction.prototype.perform = function () {
-            var self = this,
-                navigateTo = this.domainObject,
-                currentObject = self.navigationService.getNavigation();
-
-            function allow() {
-                var navigationAllowed = true;
-                self.policyService.allow("navigation", currentObject, navigateTo, function (message) {
-                    navigationAllowed = self.$window.confirm(message + "\r\n\r\n" +
-                        " Are you sure you want to continue?");
-                });
-                return navigationAllowed;
-            }
-
-            function cancelIfEditing() {
-                var editing = currentObject.hasCapability('editor') &&
-                    currentObject.getCapability('editor').isEditContextRoot();
-
-                return self.$q.when(editing && currentObject.getCapability("editor").finish());
-            }
-
-            function navigate() {
-                return self.navigationService.setNavigation(navigateTo);
-            }
-
-            if (allow()) {
-                return cancelIfEditing().then(navigate);
-            } else {
-                return this.$q.when(false);
+            if (this.navigationService.shouldNavigate()) {
```

Comment:
Using an asynchronous navigation results in a tiny amount of latency which I didn't like, but there's always the async navigate action which is useful for chaining actions.  There's a reasonable difference between the two.

I think there's a possibility to change it in the future and don't think it would be hard to change, happy to re-evaluate if we start putting setNavigation calls all over the place.  At this point I was happy to cut out some code and make it easier to follow the behavior.